tools/cli/lib/api.c

```diff
@@ -698,9 +907,53 @@ TDNFCliRepoQueryCommand(
     dwError = TDNFCliParseRepoQueryArgs(pCmdArgs, &pRepoqueryArgs);
     BAIL_ON_CLI_ERROR(dwError);
 
+    if (pRepoqueryArgs->pszQueryFormat)
+    {
+        pszQueryFormat = pRepoqueryArgs->pszQueryFormat;
+        pszQueryFormat_len = strlen(pszQueryFormat);
+    }
+
+    while (pszQueryFormat && length < pszQueryFormat_len) {
+        if (*pszQueryFormat == '%' && pszQueryFormat[1] == '{') {
+            const char *t = &pszQueryFormat[2];
+            char *tag;
+            size_t tag_len;
+            while(*t && *t != '}') t++;
+
+            if (!*t)
+                return -1;
+            tag_len = t - &pszQueryFormat[2];
+            tag = (char *)malloc(tag_len+1);
+            strncpy(tag, &pszQueryFormat[2], tag_len);
+            tag[tag_len] = 0;
+
+            if (strcmp(tag, "provides") == 0) {
```

Comment:
We have more than these 5 keys ("provides", ...) . See `depKeys` in `tools/cli/lib/parserepoqueryargs.c`.

Reply:
Added other depkeys also

Reply:
Instead of having so many `else if`'s, can we loop through them?